lab-joy/src/main.js

```diff
@@ -0,0 +1,119 @@
+import './styles/main.scss';
+import React from 'react';
+import ReactDom from 'react-dom';
+import superagent from 'superagent';
+
+const API_URL = 'https://www.reddit.com/r';
+
+class SearchForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      search: '',
+      limit: 10,
+    };
+    this.handleChange = this.handleChange.bind(this);
+    this.handleLimitChange = this.handleLimitChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleChange(e) {
+    this.setState({ search: e.target.value });
+  }
+
+  handleLimitChange(e) {
+    this.setState({ limit: e.target.value });
+  }
+
+  handleSubmit(e) {
+    e.preventDefault();
+    this.props.update_state(this.state.search, this.state.limit);
+  }
+
+  render() {
+    return (
+      <form className="search-form" onSubmit={this.handleSubmit}>
+        <h1>Search Reddit</h1>
+        <input type="text" name="search" value={this.state.search} onChange={this.handleChange} placeholder="Search" />
+        <input type="number" name="limit" min="1" max="99" value={this.state.limit} onChange={this.handleLimitChange} placeholder="#" />
+        <button type="submit">Search</button>
+      </form>
+    );
+  }
+}
+
+class Results extends React.Component {
+  constructor(props) {
+    super(props);
+  }
+
+  render() {
+    return (
+      <div className="results">
+        {this.props.search ?
+          <section className="search-data">
+            <ul>{this.props.search.data.children.map((el, i) => {
+              return (
+                <li>
```

Comment:
you need to be passing the key in li like this <li key={i}>  because with the way it is now I'm getting unique key warnings in the console.